### PR TITLE
Shell Sort in Python

### PR DESCRIPTION
**Pull request by narayan2111**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

It is a generalized version of insertion sort.
In shell sort, elements at a specific interval are sorted.

Sample I/O
	Input:
	
	Enter Size : 5
	Enter Elements: 5 4 3 2 1

	Output:
	Array After Sorting: 1 2 3 4 5

Time Complexity: O(n*2)
Space Complexity: O(1)

### Checklist

- [ Y] I've read the contribution guidelines.
- [ Y] I've checked the issue list before deciding what to submit.
- [ Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#387 
ProgramOutput

### CODE REVIEW

**Comment by narayan2111** [comment]
Kindle review this PR
@SKAUL05 @iamrajiv @Kajol-Kumari @Abhijit2505

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@narayan2111 please follow PEP 8 rules to format the code.

**Comment by narayan2111** [comment]
> @narayan2111 please follow PEP 8 rules to format the code.

Done Sir

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@narayan2111 still it does not follow PEP 8 rules you can check your program here http://pep8online.com/

**Comment by narayan2111** [comment]
> @narayan2111 still it does not follow PEP 8 rules you can check your program here http://pep8online.com/

Sir, code works fine there but here it doesn't work fine, now, what should I do to pass this test case?

File: Python/sort/Shell_Sort.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+print ('Enter elements:')
+for i in range(0, size):
+    item = int(input())
+    arr.append(item)
+
+shellSort(arr)
+
+print ('\nSorted Array:')
+for i in range(size):
+    print (arr[i])
```
Comment (atarax665) [inline Python/sort/Shell_Sort.py:40]: Add sample input/output in the end in commented form.

File: Python/README.md [hunk 2]
```diff
@@ -1,4 +1,407 @@
...
+* [Circular Linked List](ds/Circular_LinkedList.py)
+* [Infix To Postfix Conversion](ds/Infix_To_Postfix_Conversion.py)
+* [Sparse Matrix ](ds/Sparse_matrix_Operation.py)
+
+## Searching
+* [Bilinear Search](search/bilinear.py)
+* [Binary Search](search/binarysearch.py)
+* [Breadth First Search](search/bfs.py)
+* [Depth First Search](search/dfs.py)
+* [Interpolation Search](search/Interpolation_Search.py)
```
Comment (atarax665) [inline Python/README.md]: Revert the changes and just add hyperlink of your program in the redme file.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/sort/Shell_Sort.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+def shellSort(arr):
+
+    # Start with a big gap, then reduce the gap
+
+    size = len(arr)
+    gap = size // 2
+
+    # Do a gapped insertion sort for this gap size.
+
```
Comment (vatsalkesarwani12) [inline Python/sort/Shell_Sort.py]: remove all extra spaces

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

**Review by atarax665 — CHANGES_REQUESTED** [review]
The readme file still contains repetitive blocks. Rever to other merged PRs to get a clear idea of what you are supposed to edit.

**Comment by narayan2111** [comment]
> The readme file still contains repetitive blocks. Rever to other merged PRs to get a clear idea of what you are supposed to edit.

Updated the readme file @atarax665

**Comment by vatsalkesarwani12** [comment]
remove branch conflict

**Comment by narayan2111** [comment]
> remove branch conflict

Resolved
Kindly review this PR @atarax665 @iamrajiv @plazzy99

**Review by vatsalkesarwani12 — APPROVED** [review]

**Review by iamrajiv — APPROVED** [review]

**Comment by narayan2111** [comment]
Kindly look to this PR @atarax665

**Review by atarax665 — APPROVED** [review]
LGTM :)

**Comment by narayan2111** [comment]
> LGTM :)

Kindly merge this PR sir. :)

**Pull request merged**